students/hirot_directory/session04/trigrams_sherlock_small.py

```diff
@@ -0,0 +1,72 @@
+
+"""
+Read sherlock_small.txt
+Create the function of trigrams
+"""
+
+import random
+
+
+def read_file():
+
+    lines = []
+
+    with open('sherlock_small.txt', 'r') as f:
+        while True:
+            line = f.readline().split()
+            lines.extend(line)
+            if not line:
+                break
+            # print(line)
```

Comment:
Note using `if not line` to stop your loop means that the loop will stop at the first blank line. This is probably not what you want. We were able to use this syntax to read in whole files earlier, because we were not processing the line before testing it. Once you use split, you are getting rid of the command keys (end of line characters), so there is no longer any difference between an empty line and the final line. You may have to look at the text file to see if there is something else you can use as a stopping point. There is an obvious stopping point in the big file, but if you are using the small one, you may have to get more creative, or check for the end before processing the line, but this adds more complexity.

Reply:
You know, if you use the `for line in lines:` idiom instead of a while loop, then you will just get all of the lines in  the file, and you don't have to worry about figuring out the end of the file. That being said, for the big file, there is a bunch of stuff at the end that you don't want, so you may still prefer to use custom endpoint.